### PR TITLE
Add ac partner.v3 support

### PR DESCRIPTION
**Pull request by VLGorskij**

### CODE REVIEW

**Comment by Apollon77** [comment]
Why your PR change copyright and remove changelog? SOrry but so it is not mergeable

**Comment by VLGorskij** [comment]
Good day! I took into account your comments and corrected the PR.02.08.2020, 21:43, "Ingo Fischer" <notifications@github.com>:
Why your PR change copyright and remove changelog? SOrry but so it is not mergeable

File: README.md [hunk 1]
```diff
@@ -1,90 +1,105 @@
...
+  
+  The following files have been modified with respect to the official version of the adapter:
+  - main.js
+  - hub.js
+  - Gateway.js
+  - devices.js
+  - THSensor.js
+  
+  
+## Описание
```
Comment (Apollon77) [inline README.md:36]: Please have only english in README. A readme in other language can be created in a "docs/ru/README.md" folder and then linked.

File: README.md [hunk 2]
```diff
@@ -1,90 +1,105 @@
...
-- `sensor_motion.aq2` - Xiaomi Aqara Motion Sensor
-- `sensor_wleak.aq1` -  Xiaomi Aqara water sensor
-- `ctrl_ln2.aq1` -      Xiaomi Aqara Wall Switch LN (Double)
-- `remote.b286acn01` -  Xiaomi Aqara Wireless Remote Switch (Double Rocker)
-- `remote.b1acn01` -    Xiaomi Aqara Wireless Remote Switch
-- `vibration` -         Xiaomi vibration Sensor
-- `wleak1` -            Xiaomi Aqara Water Sensor
-- `lock_aq1` -          Xiaomi Lock
-- `relay.c2acn01` -     Aqara 2 channels relay control module (__using SID number__)
+## English/Русский
```
Comment (Apollon77) [inline README.md:4]: Only one language please ... more details see below

File: lib/Sensors/Gateway.js [hunk 3]
```diff
@@ -48,31 +56,53 @@ Gateway.prototype.getData = function (data) {
         obj.proto_version = this.protoVersion;
         newData           = true;
     }
-
-    if ((data.ac_state !== undefined) && (data.ac_state !== this.acState)) {
```
Comment (Apollon77) [inline lib/Sensors/Gateway.js:52]: This is not needed anymore?

File: lib/Sensors/THSensor.js [hunk 4]
```diff
@@ -61,10 +61,10 @@ THSensor.prototype.getData = function (data) {
         this.humidity = parseInt(data.humidity) / 100;
         obj.humidity = this.humidity;
         newData = true;
-    }
-
+    }    
     if (data.pressure !== undefined && data.pressure !== null && data.pressure !== 0 && data.pressure !== '0') {
-        this.pressure = parseInt(data.pressure) / 100;
+        this.pressure = Math.round(parseInt(data.pressure) / 133.322);      // Pressure in mmHg
```
Comment (Apollon77) [inline lib/Sensors/THSensor.js:66]: You change the value? Was is broken before?

File: lib/devices.js [hunk 5]
```diff
@@ -81,23 +81,34 @@ const states = {
...
+    },
+    // Added new states for acpartner3
+    acpartner3:             {type: 'acpartner.v3',    fullName: 'Xiaomi Aqara AC partner Gateway', ClassName: Gateway, states: {
+            illumination:   {name: 'Illumination',    role: 'value.lux',         write: false, read: true,  type: 'number', unit: 'lux'},
+            mid:            {name: 'Music ID',        role: 'state',             write: true,  read: false, type: 'number', desc: '10000 - stop, 10005 - custom ringtone'},
+            on_off_cfg:     {name: 'Air conditioning switch status',    role: 'state',          write: true, read: true,  type: 'string'},
+            mode_cfg:       {name: 'Air conditioning mode',             role: 'state',          write: true, read: true,  type: 'string'},
+            ws_cfg:         {name: 'Air conditioning speed',            role: 'state',          write: true, read: true,  type: 'string'},
+            swing_cfg:      {name: 'Air conditioning swing',            role: 'state',          write: true, read: true,  type: 'string'},
+            temp_cfg:       {name: 'Air conditioning temperature',      role: 'state',          write: true, read: true,  type: 'string'},
```
Comment (Apollon77) [inline lib/devices.js:104]: why not "level.temperature" as role?

File: lib/Sensors/Gateway.js (hunk 3 above)
Comment (VLGorskij) [reply lib/Sensors/Gateway.js:52]: Yes, it is no longer needed. ac_state carries little useful information

File: lib/Sensors/THSensor.js (hunk 4 above)
Comment (VLGorskij) [reply lib/Sensors/THSensor.js:66]: No, everything worked. Here the conversion of units of measurement from hectopascals to mm Hg

File: lib/devices.js (hunk 5 above)
Comment (VLGorskij) [reply lib/devices.js]: changed

**Comment by VLGorskij** [comment]
Hello! I read your comments and made the appropriate corrections.

File: lib/Sensors/Gateway.js [hunk 6]
```diff
@@ -14,7 +14,14 @@ function Gateway(sid, ip, hub, model) {
     this.illumination = null;
     this.protoVersion = '';
     this.acPower      = undefined;
-    this.acState      = undefined;
+	// New states for acpartner-----------------
+    this.OnOffCfg     = undefined;
```
Comment (GermanBluefox) [inline lib/Sensors/Gateway.js:18]: Please use only low case names like onOffCfg. Capital letters used in classes.

Comment (VLGorskij) [reply lib/Sensors/Gateway.js]: > Please use only low case names like onOffCfg. Capital letters used in classes.

ok! Corrected.

Comment (VLGorskij) [reply lib/Sensors/Gateway.js]: ok! Corrected.

**Comment by VLGorskij** [comment]
use only low case names

**Comment by VLGorskij** [comment]
Good day! I edited the readme file. As for the pressure units (...It must be configurable in settings), I don't know how to do this yet. Maybe you can help?

**Comment by VLGorskij** [comment]
> And please check the readme file, because I modified it to have description for gateway and for new acpartner too

Good day! I edited the readme file. As for the pressure units (...It must be configurable in settings), I don't know how to do this yet. Maybe you can help?

**Pull request merged**